### DISCUSSION TITLE
Actor default state after construction

### DISCUSSION THREAD

**Discussion by HermansG** [akkadotnet/akka.net#7416] [Q&A] [answered]
When an Actor is created it has a certain State. In de ctor one can define e.g. some Receive's.
Shouldn't this State have a certain name or status?

**Comment by Aaronontheweb** [comment, marked as answer]
In an `UntypedActor`, the default behavior is going to be your `OnReceive` method - the `abstract` method that has to be filled in. All actor base classes with the notable exception of the `FSM<TState>` have a fairly loose and unopinionated view of what constitutes a state. 

I'd define "actor state" as your actor's:

1. Internal data fields / properties
2. Current message processing behavior

**Reply by HermansG** [reply]
I was only thinking about ReceiveActor's.
After the ctor you will have 1 & 2 it is _as if_ the actor implicitly executed Become(DefaultState)
Probably a personal preference or a didactical preference

**Reply by Aaronontheweb** [reply]
For `ReceiveActor`s I usually create a method that has all of my "default" receives in it and just call that from the CTOR - that makes it a little more natural to do behavior-switching that way
